From 30251d7754fe308251747dc40328a52a6b0e0601 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Sat, 8 Aug 2026 00:27:15 +0300
Subject: [PATCH] thd_gddv: fix building on 32-bit systems

Fix the error caused by type mismatch:

src/thd_gddv.cpp: In member function 'int cthd_gddv::evaluate_condition(condition&)':
src/thd_gddv.cpp:1721:39: error: format '%ld' expects argument of type 'long int', but argument 4 has type 'time_t' {aka 'long long int'} [-Werror=format=]

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Upstream-Status: Submitted [https://github.com/intel/thermal_daemon/pull/596]
---
 src/thd_gddv.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/thd_gddv.cpp b/src/thd_gddv.cpp
index e822b9f72f45..a2691594960d 100644
--- a/src/thd_gddv.cpp
+++ b/src/thd_gddv.cpp
@@ -1718,7 +1718,7 @@ int cthd_gddv::evaluate_condition(struct condition& condition) {
 			return ret;
 
 		if (condition.time) {
-			thd_log_debug("time condition matched %ld \n", condition.state_entry_time);
+			thd_log_debug("time condition matched %jd \n", (intmax_t)condition.state_entry_time);
 			if (condition.state_entry_time == 0) {
 				condition.state_entry_time = time(nullptr);
 				return THD_ERROR;
-- 
2.47.3

